mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,314 @@
+                elif 'Update Rate (Hz)' in i_line:
+                    srate = float(i_line.rsplit(' ')[0])
+                elif 'Updata Rate (Hz)' in i_line:
+                    # Version 0.40 of the BOXY recording software
+                    # (and possibly other versions lower than 0.84) contains a
+                    # typo in the raw data file where 'Update Rate' is spelled
+                    # "Updata Rate. This will account for this typo.
+                    srate = float(i_line.rsplit(' ')[0])
+                elif '#DATA BEGINS' in i_line:
+                    # Data should start a couple lines later.
+                    start_line = line_num + 2
+                if start_line > 0 & end_line == 0:
+                    if line_num == start_line - 1:
+                        # Grab names for each column of data.
+                        col_names = np.asarray(re.findall(
+                            r'\w+\-\w+|\w+\-\d+|\w+', i_line.rsplit(' ')[0]))
+                        if 'exmux' in col_names:
+                            # Change filetype based on data organisation.
+                            filetype = 'non-parsed'
+                        if 'digaux' in col_names:
+                            mrk_col = np.where(col_names == 'digaux')[0][0]
+                    # Need to treat parsed and non-parsed files differently.
+                    elif (mrk_col > 0 and line_num > start_line and
+                          filetype == 'non-parsed'):
+                        # Non-parsed files have different lines lengths.
+                        crnt_line = i_line.rsplit(' ')[0]
+                        temp_data = re.findall(r'[-+]?\d*\.?\d+', crnt_line)
+                        if len(temp_data) == len(col_names):
+                            mrk_data.append(float(
+                                re.findall(r'[-+]?\d*\.?\d+', crnt_line)
+                                [mrk_col]))
+                    elif (mrk_col > 0 and line_num > start_line
+                          and filetype == 'parsed'):
+                        # Parsed files have the same line lengths for data.
+                        crnt_line = i_line.rsplit(' ')[0]
+                        mrk_data.append(float(
+                            re.findall(r'[-+]?\d*\.?\d+', crnt_line)[mrk_col]))
+
+        # Label each channel in our data, for each data type (DC, AC, Ph).
+        # Data is organised by channels x timepoint, where the first
+        # 'source_num' rows correspond to the first detector, the next
+        # 'source_num' rows correspond to the second detector, and so on.
+        boxy_labels = list()
+        for det_num in range(detect_num):
+            for src_num in range(source_num):
+                for i_type in ['DC', 'AC', 'Ph']:
+                    boxy_labels.append('S' + str(src_num + 1) +
+                                       '_D' + str(det_num + 1) + ' ' + i_type)
+
+        # Determine channel types.  
```

Comment:
@rob-luke  ok I've added the following back up to line 143

Reply:
The second type should probably be `fnirs_fd_ac_amplitude`?

Reply:
Below you refer to

```python
chans_dc
chans_ac
chans_ph
```

so is there a type missing here?

Reply:
yes now assign all three:
```
    # Determine channel types.  
        if datatype == 'Ph':    
            chan_type = 'fnirs_fd_phase'    
        elif datatype == 'DC':
            chan_type = 'fnirs_fd_dc_amplitude'
        else:   
            chan_type = 'fnirs_fd_ac_amplitude'   ```